src/Microsoft.AspNetCore.Mvc.Formatters.Xml/ValidationProblemDetailsWrapper.cs

```diff
@@ -0,0 +1,132 @@
+    public sealed class ValidationProblemDetailsWrapper : IXmlSerializable, IUnwrappable
+    {
+        private static readonly string EmptyKey = SerializableErrorWrapper.EmptyKey;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ValidationProblemDetailsWrapper"/>.
+        /// </summary>
+        public ValidationProblemDetailsWrapper()
+            : this(new ValidationProblemDetails())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ValidationProblemDetailsWrapper"/> for the specified
+        /// <paramref name="problemDetails"/>.
+        /// </summary>
+        /// <param name="problemDetails">The <see cref="ValidationProblemDetails"/>.</param>
+        public ValidationProblemDetailsWrapper(ValidationProblemDetails problemDetails)
+        {
+            ValidationProblemDetails = problemDetails;
+        }
+
+        /// <summary>
+        /// Gets the wrapped <see cref="Mvc.ValidationProblemDetails"/>.
+        /// </summary>
+        public ValidationProblemDetails ValidationProblemDetails { get; }
+
+        /// <inheritdoc />
+        public XmlSchema GetSchema() => null;
+
+        /// <inheritdoc />
+        public void ReadXml(XmlReader reader)
+        {
+            if (reader.IsEmptyElement)
+            {
+                reader.Read();
+                return;
+            }
+
+            reader.ReadStartElement();
+            while (reader.NodeType != XmlNodeType.EndElement)
+            {
+                var key = XmlConvert.DecodeName(reader.LocalName);
+                if (key == nameof(ValidationProblemDetails.Errors))
+                {
+                    reader.Read();
+                    ReadErrorProperty(reader);
+                }
+                else
+                {
+                    ProblemDetailsWrapper.ReadProperty(reader, ValidationProblemDetails, key);
+                }
+
+                reader.MoveToContent();
+            }
+
+            reader.ReadEndElement();
+        }
+
+        private void ReadErrorProperty(XmlReader reader)
+        {
+            if (reader.IsEmptyElement)
+            {
+                return;
+            }
+
+            while (reader.NodeType != XmlNodeType.EndElement)
+            {
+                var key = XmlConvert.DecodeName(reader.LocalName);
+                var value = reader.ReadInnerXml();
+                if (string.Equals(EmptyKey, key, StringComparison.Ordinal))
+                {
+                    key = string.Empty;
+                }
+
+                ValidationProblemDetails.Errors.Add(key, new[] { value });
+                reader.MoveToContent();
+            }
+        }
+
+        /// <inheritdoc />
+        public void WriteXml(XmlWriter writer)
+        {
+            ProblemDetailsWrapper.WriteProblemDetails(writer, ValidationProblemDetails);
+            writer.WriteStartElement(XmlConvert.EncodeLocalName(nameof(ValidationProblemDetails.Errors)));
```

Comment:
Need a more unique element name. This conflicts with adding an `"Errors"` entry to `ProblemDetails.ExtensionMembers` and information intended for one property will round-trip into the other.

Reply:
My slight concern is that it breaks the with 2.1. I had a look at this type and it's kinda hard to pivot this feature on CompatSwitch (since it's not from DI and we often new instances of it and expect it to be correctly configured). If we're 🆗 with breaking the format, the spec actually has a schema that we don't use today: https://tools.ietf.org/html/rfc7807#appendix-A

Reply:
This `class` is new and I'm not clear how `Errors` could have been serialized without it.

Using the XML schema would be an interesting approach. You're not that far off at the moment. But, would still need a way to distinguish the `Errors` members from bits within an `ExtensionMembers["Errors"]` entry.